internal/provider/assessor/openai_assessor_edge_test.go

```
@@ -0,0 +1,877 @@
+		}))
+	}))
+	defer srv.Close()
+
+	v := NewOpenAIAssessor(newTestVerifierConfig(srv.URL, "key", "assessor-model"), srv.Client())
+	v.SetMetrics(metrics)
+	ctx := observability.WithAIOperation(context.Background(), observability.AIOperationPlacementAssessment, 1)
+	_, err := v.openAIStructuredChatJSONWithUsage(ctx, "assessor-model", "schema", map[string]any{}, "system", map[string]any{})
+	require.ErrorIs(t, err, ErrVerifierProvider)
+
+	recorder := httptest.NewRecorder()
+	metrics.Handler().ServeHTTP(recorder, httptest.NewRequest(http.MethodGet, "/metrics", nil))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)